app/controllers/application_controller.rb

```diff
@@ -27,6 +27,10 @@ def admin_required
     redirect_to root_path unless current_user && current_user.admin?
   end
 
+  # To use in before_action with parameters, do
+  # before_action do
+  #   allowed_feature_required(...)
+  # end
   def allowed_feature_required(allowed_feature, allow_admin = false)
     redirect_to root_path unless current_user && (
```

Comment:
This is old but... should we redirect to root here? Why not just an 404 or something like that or maybe add a parameter for that?

Reply:
I believe we discussed offline that the downside of returning 403 is that it would indicate that there is something hidden at that URL. Right now, if I type in a random url that doesn't match anything, it redirects to home for some reason. Will defer addressing for now.